cache/redis_cache.go

```diff
@@ -240,16 +247,23 @@ func (r *redisCache) decodeMetadata(b []byte) (*ContentMetadata, int, error) {
 
 func (r *redisCache) Put(reader io.Reader, contentMetadata ContentMetadata, key *Key) (time.Duration, error) {
```

Comment:
I wonder, should we have a timeout on entire PUT? Meaning for everything, max timeout is...

Reply:
this is what we had before the PR. I removed it because it could failed if clickhouse has long running queries (queries that takes multiple seconds btw the first batch of data and the last batch). My guess is that we were facing this pb in CS production.

Reply:
indeed, forgot about this case!